testdata/v2_atlantis_depends_on/atlantis.yaml

```diff
@@ -0,0 +1,35 @@
+# Auto-generated by fogg. Do not edit
+# Make improvements in fogg, so that everyone can benefit.
+
+version: 3
+projects:
+  - name: test_db
+    dir: terraform/envs/test/db
+    workspace: default
+    terraform_version: 0.100.0
+    autoplan:
+      when_modified:
+        - '*.tf'
+        - ../../../../terraform/modules/my_module/**/*.tf
+        - ../../../../terraform/modules/my_module/**/*.tf.json
+      enabled: true
+    apply_requirements:
+      - approved
+  - name: test_vpc
+    dir: terraform/envs/test/vpc
+    workspace: default
+    terraform_version: 0.100.0
+    autoplan:
+      when_modified:
+        - '*.tf'
+        - '!remote-states.tf'
+        - ../../../../terraform/modules/my_module/**/*.tf
+        - ../../../../terraform/modules/my_module/**/*.tf.json
+        - ../../../../foo_modules/bar/**/*.tf
+        - ../../../../foo_modules/bar/**/*.tf.json
```

Comment:
LGTM, only the vpc project ignores remote states. the db project does not because it has `depends_on>components` array and `autoplan_remote_states` is explicitly set to `true`